eclipse/che-theia
# Pin to upstream Theia commit as adaptation is required

Signed-off-by: Artem Zatsarynnyi <azatsary@redhat.com>



### What does this PR do?
Stabilizes Che Theia build and fixes critical runtime issues by:
- switching to upstream Theia [commit](https://github.com/eclipse-theia/theia/commit/dc917ee076b206e2b4b3c24869aa55d2d1962037) as an adaptation to the latest changes is required - [`SemanticTokens` Plug-in API](https://github.com/eclipse-theia/theia/commit/d53cbaabce9b52747eeb591e0d834766bdfe6fca)
- switching to the previous version of `@eclipse-che/workspace-client` as the latest one [brings breaking changes](https://github.com/eclipse/che-workspace-client/pull/39)

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

closes https://github.com/eclipse/che/issues/18255 and fixes partially https://github.com/eclipse/che/issues/18278

### How to test this PR?

1. Use `How to test` button, provided by @che-bot, to spin-up the Workspace.
2. Check that the plug-ins are loaded.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable
